internal/provider/integration_azure_resource.go

```diff
@@ -0,0 +1,260 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	mondoov1 "go.mondoo.com/mondoo-go"
+)
+
+var _ resource.Resource = (*integrationAzureResource)(nil)
+
+func NewIntegrationAzureResource() resource.Resource {
+	return &integrationAzureResource{}
+}
+
+type integrationAzureResource struct {
+	client *ExtendedGqlClient
+}
+
+type integrationAzureResourceModel struct {
+	// scope
+	SpaceId types.String `tfsdk:"space_id"`
+
+	// integration details
+	Mrn                    types.String `tfsdk:"mrn"`
+	Name                   types.String `tfsdk:"name"`
+	ClientId               types.String `tfsdk:"client_id"`
+	TenantId               types.String `tfsdk:"tenant_id"`
+	SubscriptionsWhitelist types.List   `tfsdk:"subscription_whitelist"`
```

Comment:
The terms are deprecated for the backend, we should expose `SubscriptionAllowList` and `SubscriptionDenyList`